### DISCUSSION TITLE
RPi3b: Langsames Laden der Device-Status Infos

### DISCUSSION THREAD

**Discussion by jp112sdl** [OpenCCU/OpenCCU#1844] [🇩🇪 Generelle Diskussionen]
Der Aufruf von _Einstellungen>Geräte_ zeigt sofort eine Liste aller Geräte an.
Im Hintergrund werden die RSSI-Werte so "halb-asynchron" nachgeladen.

Während dieser Ladezeit bleibt der Hintergrund-Eventlistener stehen ("Heartbeat-Blinken der Himbeere").
Rufe ich innerhalb der Ladezeit schon mal die Einstellungsseite eines Geräts auf, wird diese erst vollständig geladen, wenn die eben erwähnte Hintergrundaktualisierung fertig ist.

Genau genommen dauert es bei mir im Produktivsystem mit 113 Geräten knapp **11 Sekunden**, bis vom letzten Gerät die RSSI-Werte geladen wurden:
<img width="546" alt="Bildschirmfoto 2022-05-22 um 19 44 45" src="https://user-images.githubusercontent.com/18190139/169708502-92379d1b-25dc-42e0-8e12-80a0e6806a93.png">

Im Testsystem mit 54 Geräten dauert es rund die Hälfte der Zeit.
Es lässt sich also direkt von der Anzahl der Geräte abhängig machen.

Es ist total anstrengend, wenn man in die Einstellungen von einem Gerät zum nächsten möchte.

Geht es nur mir so?
Hat jemand eine Idee, wie/was man da optimieren könnte?

P.S.: Liegt es gar nicht an den RSSI Werten sondern an der Abfrage der Interface-Zugehörigkeit? 😮

**Comment by blackbasket** [comment]
Ich habe es gerade bei mir mal nachgestellt...

Safari (iPadOS) mit RM auf rpi3 ("Original-"CCU3) und 3.61.7.20220115 nachgestellt... 56 Geräte...

Da kann ich es nicht reproduzieren...

**Reply by jp112sdl** [reply]
> 56 Geräte

@blackbasket Hast du BidCos-Geräte? Falls ja - wie viele?

**Reply by blackbasket** [reply]
Das sind nur 3...

**Reply by jp112sdl** [reply]
@blackbasket Ok, dann hab ich eine Ahnung, was da bei mir so ewig dauert und bestätigt meinen unter "P.S." erwähnten verdacht -,-

**Reply by blackbasket** [reply]
Ja, wäre definitiv möglich... meinen Hinweis auf die alte FW-Version hattest du wahrgenommen?

**Reply by jp112sdl** [reply]
Ja, hab ich. Das beschriebene Verhalten wurde auch mit genau dieser Version erst eingeführt. 
Ich hatte da den `0161-WebUI-Add-ShowRFGatewayInDeviceList` Patch eingebaut. 
Muss mal schauen, ob bzw. wie sich da was optimieren lässt

**Reply by jp112sdl** [reply]
Es liegt nicht am `0161-WebUI-Add-ShowRFGatewayInDeviceList` Patch, sondern doch am Laden der RSSI-Info.
Wenn ich den Block
https://github.com/jens-maus/RaspberryMatic/blob/323203392ee1fdb16fab98f03d1d69b1c07bd2a7/buildroot-external/patches/occu/0038-WebUI-DeviceOverview-StatusColumn.patch#L360-L371
auskommentiere, lädt die Geräteliste flüssig.

Damit kann ich leben.

Ist dann also (für mich) gelöst
